exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/Predicates.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.prometheus;
+
+import java.util.function.Predicate;
+
+public final class Predicates {
```

Comment:
I say we omit these predicates. Reasons:
 
- They're syntactic sugar but only make things the slightest bit easier
- They're not easily discoverable. If you do happen to find a `Predicates` class in the prometheus package, its not clear what its purpose is.

Reply:
The existence of these methods makes it very easy to read and understand.

`true, /* allowedResourceAttributesFilter= */ Predicates.is("cluster"));` is harder than `true, /* allowedResourceAttributesFilter= */ attributeKey -> attributeKey.equals("cluster"));`

They are only for the purpose of testing the Prometheus exporter classes: `PrometheusMetricReader` and `Otel2PrometheusConverter`.

I think once you modify a test of the classes above, you will read the existing tests and then discover it.

Reply:
What about this:

- Rename Predicates to something more domain specific, add javadoc to point to PrometheusHttpServerBuilder#setAllowedResourceAttributesFilter
- Link to it from the PrometheusHttpServerBuilder#setAllowedResourceAttributesFilter javadoc

Reply:
To ensure I follow correctly: Would you like me to move `Predicates` (after I rename it) from `src/test/java` to `src/main/java`?

Reply:
I think I misunderstood. I was under the impression this was already in `src/main/java`. I'm fine with keeping this as is.